cloudinit/sources/DataSourceOpenStack.py

```diff
@@ -219,8 +220,9 @@ def read_metadata_service(base_url, ssl_details=None,
 
 def detect_openstack(accept_oracle=False):
     """Return True when a potential OpenStack platform is detected."""
-    if not util.is_x86():
-        return True  # Non-Intel cpus don't properly report dmi product names
+    uname_arch = os.uname()[4]
+    if not (util.is_x86() or uname_arch == 'aarch64'):
+        return True
```

Comment:
I don't like this change as it will break arm64 systems that actually properly passed through.
Better would be to check and see.

The real thing we need to support is having cloud-init do the right thing on OpenStack .  when we first started using ds-identify , I wrote some bugs against openstack-nova and tagged them with [dsid](https://bugs.launchpad.net/nova/+bugs?field.searchtext=&orderby=-importance&search=Search&field.status%3Alist=NEW&field.status%3Alist=OPINION&field.status%3Alist=INVALID&field.status%3Alist=WONTFIX&field.status%3Alist=EXPIRED&field.status%3Alist=CONFIRMED&field.status%3Alist=TRIAGED&field.status%3Alist=INPROGRESS&field.status%3Alist=FIXCOMMITTED&field.status%3Alist=FIXRELEASED&field.status%3Alist=INCOMPLETE_WITH_RESPONSE&field.status%3Alist=INCOMPLETE_WITHOUT_RESPONSE&assignee_option=any&field.assignee=&field.bug_reporter=&field.bug_commenter=&field.subscriber=&field.structural_subscriber=&field.tag=dsid&field.tags_combinator=ANY&field.has_cve.used=&field.omit_dupes.used=&field.omit_dupes=on&field.affects_me.used=&field.has_patch.used=&field.has_branches.used=&field.has_branches=on&field.has_no_branches.used=&field.has_no_branches=on&field.has_blueprints.used=&field.has_blueprints=on&field.has_no_blueprints.used=&field.has_no_blueprints=on).

We need to support openstack somehow on those platforms.